src/prompto/rephrasal.py

```diff
@@ -0,0 +1,389 @@
+            for the LLM(s) for rephrasal. Each dictionary will contain a
+            new prompt for rephrasal for each input prompt in the
+            input_prompts list using the template_prompt
+        """
+        if not out_filepath.endswith(".jsonl"):
+            raise ValueError("out_filepath must end with '.jsonl'")
+
+        rephrase_prompts = self.create_rephrase_inputs(rephrase_model=rephrase_model)
+
+        logging.info(f"Creating rephrase experiment file at {out_filepath}...")
+        with open(out_filepath, "w", encoding="utf-8") as f:
+            for j_input in tqdm(
+                rephrase_prompts,
+                desc=f"Writing rephrase prompts to {out_filepath}",
+                unit="prompts",
+            ):
+                json.dump(j_input, f)
+                f.write("\n")
+
+        return rephrase_prompts
+
+    @staticmethod
+    def _convert_rephrased_prompt_dict_to_input(rephrased_prompt: dict) -> dict:
+        """
+        Method to convert a completed rephrased prompt dictionary to an input prompt dictionary.
+        This is done by:
+        - Renaming the "response" key to "prompt" (as this is the new rephrased prompt)
+        - Keep "id" key as is
+        - Keep "input-prompt" and "input-id" keys as is
+        - For all remaining keys starting with "input-", remove the "input-" prefix.
+          This may overwrite existing keys in the rephrased prompt dictionary
+          (e.g. "input-api", "input-model_name", "input-parameters"
+          should override existing keys "api", "model_name", "parameters")
+
+        Parameters
+        ----------
+        rephrased_prompt : dict
+            A dictionary containing the rephrased prompt. Should usually contain
+            the keys "id", "prompt", "input-prompt" and "input-id". Should also
+            contain "input-api", "input-model_name" and "input-parameters" keys
+
+        Returns
+        -------
+        dict
+            A dictionary containing the input prompt for a model after rephrasing.
+            The prompt will be the rephrased prompt, and there will be "input-id"
+            and "input-prompt" keys to keep track of the original input prompt.
+            The "id" key will indicate the rephrased prompt id. The "api", "model_name",
+            and other keys from the original input will be restored
+        """
+        input_prompt = {
+            "id": rephrased_prompt["id"],
+            "prompt": rephrased_prompt["response"],
+            "input-prompt": rephrased_prompt["input-prompt"],
+            "input-id": rephrased_prompt.get("input-id", "MA"),
+        }
+
+        # restore the original input keys (e.g. "api", "model_name", "parameters")
+        for k, v in rephrased_prompt.items():
+            if k.startswith("input-") and k not in ["input-prompt", "input-id"]:
+                input_prompt[k[6:]] = v
+
+        return input_prompt
+
+    def create_new_input_file(
```

Comment:
I would just add here a simple check that removes perfect lowercased duplicates - we can specify that in the documentation / notebook